sdk/ai/azure-ai-inference/src/samples/java/com/azure/ai/inference/usage/TraceStreamingChatCompletionsToolCallSample.java

```diff
@@ -0,0 +1,314 @@
+        try (AutoCloseable scope = span.makeCurrent()) {
+            final List<ChatRequestMessage> messages = new ArrayList<>();
+            messages.add(new ChatRequestSystemMessage("You an assistant that helps users find flight information."));
+            messages.add(new ChatRequestUserMessage("What is the next flights from Seattle to Miami?"));
+            final GetFlightInfoFunction function = new GetFlightInfoFunction();
+
+            final IterableStream<StreamingChatCompletionsUpdate> toolCallChunks = client.completeStream(new ChatCompletionsOptions(messages).setTools(function.toolDefinitions()));
+            final ChunksMerged toolCallChunksMerged = ChunksMerged.create(toolCallChunks);
+            final ChatCompletionsToolCall toolCall = toolCallChunksMerged.asTooCall();
+            messages.add(toAssistantMessage(toolCall));
+
+            final ChatRequestToolMessage toolMessage = function.invoke(toolCall);
+            messages.add(toolMessage);
+
+            final IterableStream<StreamingChatCompletionsUpdate> modelResponseChunks = client.completeStream(new ChatCompletionsOptions(messages).setTools(function.toolDefinitions()));
+            final ChunksMerged modelResponseChunksMerged = ChunksMerged.create(modelResponseChunks);
+            System.out.println("Model response: " + modelResponseChunksMerged.content);
+            span.end();
+        } catch (Exception e) {
+            span.setAttribute("error.type", e.getClass().getName());
+            span.end();
+            throw new RuntimeException(e);
+        } finally {
+            telemetry.close();
+        }
+    }
+
+    private static OpenTelemetrySdk configureOpenTelemetry() {
+        // With the below configuration, the runtime sends OpenTelemetry data to the local OTLP/gRPC endpoint.
+        //
+        // For debugging purposes, Aspire Dashboard can be run locally that listens for telemetry data and offer a UI
+        // for viewing the collected data. To run Aspire Dashboard, run the following docker command:
+        //
+        // docker run --rm -p 18888:18888 -p 4317:18889 -p 4318:18890 --name aspire-dashboard mcr.microsoft.com/dotnet/nightly/aspire-dashboard:latest
+        //
+        // The output of the docker command includes a link to the dashboard. For more information on Aspire Dashboard,
+        // see https://learn.microsoft.com/dotnet/aspire/fundamentals/dashboard/overview
+        //
+        // For production telemetry use cases, see Azure Monitor, https://learn.microsoft.com/java/api/overview/azure/monitor-opentelemetry-exporter-readme
+        //
+        final AutoConfiguredOpenTelemetrySdkBuilder sdkBuilder = AutoConfiguredOpenTelemetrySdk.builder();
+        return sdkBuilder
+            .addPropertiesSupplier(() -> {
+                final Map<String, String> properties = new HashMap<>();
+                properties.put("otel.exporter.otlp.endpoint", "http://localhost:4317");
```

Comment:
this is the default value, you can comment it and it'd still work, made some suggestions in the https://github.com/anuchandy/azure-sdk-for-java/pull/18